rct229/rulesets/ashrae9012019/ruleset_functions/get_zone_target_baseline_system.py

```diff
@@ -0,0 +1,244 @@
+    assert_(rmd_b["buildings"][0], "Please make sure the `buildings` key exists.")
+
+    building_b = rmd_b["buildings"][0]
+    zone_conditioning_category_dict = get_zone_conditioning_category_dict(
+        climate_zone_b, building_b
+    )
+
+    (
+        list_building_area_types_and_zones_b,
+        predominant_building_area_type_b,
+        num_floors_b,
+    ) = juxtapose(
+        lambda cz, rmi: get_hvac_building_area_types_and_zones_dict(cz, rmi),
+        lambda cz, rmi: get_predominant_hvac_building_area_type(cz, rmi),
+        lambda cz, rmi: get_number_of_floors(cz, rmi),
+    )(
+        climate_zone_b, rmd_b
+    )
+
+    floor_area_b = sum(
+        [
+            list_building_area_types_and_zones_b[bat]["floor_area"]
+            for bat in list_building_area_types_and_zones_b
+        ]
+    )
+    expected_system_type_dict_b = expected_system_type_from_table_g3_1_1_dict(
+        predominant_building_area_type_b, climate_zone_b, num_floors_b, floor_area_b
+    )
+
+    is_cz_0_to_3a_result_bool = is_cz_0_to_3a_bool(climate_zone_b)
+
+    zones_and_systems_b = {
+        zone_b["id"]: expected_system_type_dict_b
+        for zone_b in find_all("$.buildings[*].building_segments[*].zones[*]", rmd_b)
+        if zone_conditioning_category_dict[zone_b["id"]]
+        in (
+            ZCC.CONDITIONED_RESIDENTIAL,
+            ZCC.CONDITIONED_NON_RESIDENTIAL,
+            ZCC.CONDITIONED_MIXED,
+        )
+    }
+
+    # go through each exception to Table G3.1.1 in order
+    # G3.1.1b
+    Is_g311b_implemented = False
+    if floor_area_b > BUILDING_AREA_40000_ft2:
+        for building_area_type in list_building_area_types_and_zones_b:
+            if building_area_type != predominant_building_area_type_b and (
+                list_building_area_types_and_zones_b[building_area_type]["floor_area"]
+                >= BUILDING_AREA_20000_ft2
+            ):
+                Is_g311b_implemented = True
+
+                secondary_system_type_b = expected_system_type_from_table_g3_1_1_dict(
+                    building_area_type,
+                    climate_zone_b,
+                    num_floors_b,
+                    floor_area_b,
+                )
+                for zone_b in zones_and_systems_b:
+                    if (
+                        zone_b
+                        in list_building_area_types_and_zones_b[building_area_type][
+                            "zone_ids"
+                        ]
+                    ):
+                        zones_and_systems_b[zone_b] = {
+                            "expected_system_type": secondary_system_type_b[
+                                "expected_system_type"
+                            ],
+                            "system_origin": SYSTEMORIGIN.G311B,
+                        }
+    if not Is_g311b_implemented:
+        for zone_b in zones_and_systems_b:
+            # G3.1.1c
+            if does_zone_meet_g3_1_1c(rmd_b, zone_b, False, zones_and_systems_b):
```

Comment:
is_leap_year needs to pass in.

This variable can also be passed in as an argument of this function.

Reply:
The third argument ```False``` is for the ```leap_year```.

Should we add one more argument for the ```leap_year``` in the ```get_zone_target_baseline_system``` function? Or should I get the ```leap_year``` from the ```Calendar``` key in TCD?

Reply:
yes we should add this additional argument to this function.
Leap year should comes from rule because it is higher than rmi level.